aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

```diff
@@ -103,12 +103,6 @@ struct cpu_scatter_gather_base_kernel {
   void operator()(Tensor& self, int64_t dim,
     const Tensor& index, const Scalar& value,
     const std::string& method_name, func_t& kernel_func) {
-    // no-op if index is empty
-    if (index.numel() == 0) {
-      return;
-    }
-
-    dim = maybe_wrap_dim(dim, self.dim());
```

Comment:
Since you are removing these `maybe_wrap_dim` calls (others included), you should call it once before calling these device specific kernels in the IMPL function for both `scatter` and `gather` kernels.

Reply:
Thanks for pointing out, should be fixed now.